backend/climateconnect_api/utility/email_setup.py

```diff
@@ -272,3 +278,117 @@ def remove_contact_from_list(contact_id, list_id):
         ]
     }
     mailjet_api.contact_managecontactslists.create(id=contact_id, data=data)
+
+
+def create_variables_for_weekly_recommendations(language_code: str, project_ids: List = [], organization_ids: List = [], idea_ids: List = []):
+    # edit the image and url path for production
+    # todo idea has no url 
+   
+    main_page = {
+        "en": "https://climateconnect.earth/en/browse",
+        "de": "https://climateconnect.earth/de/browse",
+    }
+    project_translation = {
+        "en": "Project",
+        "de": "Projekt",
+    }
+    organization_translation = {
+        "en": "Organization",
+        "de": "Organisation",
+    }
+    idea_translation = {
+        "en": "Idea",
+        "de": "Idee",
+    }
+
+    entities = []
+    for project_id in project_ids:
+        project = Project.objects.select_related('loc').get(id=project_id)
+        project_template = {
+            "type": project_translation.get(language_code, "en"),
+            "name": project.name,
+            "imageUrl": (settings.ALLOWED_HOSTS[6] + project.thumbnail_image.url) if project.thumbnail_image else '',
+            "url": (settings.FRONTEND_URL + "/projects/" + project.url_slug) if project.url_slug else main_page.get(language_code, "en"),
+            "location": project.loc.name if project.loc else '',
+            "creator": '',
+            "tags": '',
+        }
+        entities.append(project_template)
+    for organization_id in organization_ids:
+        organization = Organization.objects.select_related('location').get(id=organization_id)
+        orga_template = {
+            "type": organization_translation.get(language_code, "en"),
+            "name": organization.name,
+            "imageUrl": (settings.ALLOWED_HOSTS[6] + organization.thumbnail_image.url) if organization.thumbnail_image else '',
+            "url": (settings.FRONTEND_URL + "/organizations/" + organization.url_slug) if organization.url_slug else main_page.get(language_code, "en"),
+            "location": organization.location.name if organization.location else '',
+            "creator": '',
+            "tags": '',
+        }
+        entities.append(orga_template)
+    for idea_id in idea_ids:
+        idea = Idea.objects.select_related('location', 'user').get(id=idea_id)
+        idea_template = {
+            "type": idea_translation.get(language_code, "en"),
+            "name": idea.name,
+            "imageUrl": (settings.ALLOWED_HOSTS[6] + idea.thumbnail_image.url) if idea.thumbnail_image else '',
+            # How does the idea url generation work
```

Comment:
You need the hub it was posted in to generate an idea's url. 
It's `URL/hubs/<hubUrl>?idea=<slug>#ideas`.